### ISSUE TITLE
Replace TSLint with ESLint

### ISSUE THREAD

**Issue by giovannipessiva** [giovannipessiva/l4w#14]
TSLint has been deprecated in favor of ESLint: https://github.com/palantir/tslint/issues/4534
Should remove TSLint from the Grunt script and replace it with ESLint, maitaining the rules defined in the TSLint configuration file

**Comment by creativenull** [comment]
Hey! Mind if I do this?

**Comment by giovannipessiva** [comment]
Hi, I have assigned the issue to you, thanks!

**Comment by creativenull** [comment]
Thanks! I'm new to moving from tslint to eslint, so I googled around [to find this guide](https://code.visualstudio.com/api/advanced-topics/tslint-eslint-migration). I was successfully able to get it to lint with eslint, but there were a couple problems that was not addressed in tslint and generated errors and warning in the project:

+ Indentation - eslint likes to keep the indents to 2 spaces but all your ts files have 4 spaces of indent. There was no indication of how many spaces in `tslint.json`. So the question would be to change the eslint rule to 4 spaces? Or just turn off those errors?
+ JSdoc - eslint required couple of the docstrings to have a newline. It's just in a couple files not all, I believe it was 4 files in total. Should I turn off this warning? Or should I keep it and adjust those files accordingly?

**Comment by giovannipessiva** [comment]
- Indentation: try adding this to eslint config file (see: https://stackoverflow.com/a/48906878/2700039)
`"indent": ["error", 4]`
- JSdoc: keep the warning, I'll check the files later.

Thanks!!

**Comment by creativenull** [comment]
Ok so I adjusted the indent, I used `"@typescript-eslint/indent": ["error", 4]` as recommended instead of `indent` but I still get indent errors. Which makes sense checking those files (where spaces are 5 instead of 4), so I'm guessing it works as intended just more strict. Will this be good enough for the PR?

I tried with `indent` before and I got more errors (~800 w/ indent vs ~500 w/ typescript-indent). So it would be better to just stick with `@typescript-eslint/indent` rule instead?

> JSdoc: keep the warning, I'll check the files later.

I've changed JSdoc from error to warning 👍
